### PR TITLE
Implement calculator

### PR DESCRIPTION
**Pull request by cschiragb**

### CODE REVIEW

File: calculator.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    add_number: function (number) {
+      numbers.push(number);
+    },
+    calculate: function () {
+      // Return zero if there are no numbers to calculate
+      if (numbers.length === 0) {
+        throw 'Cannot calculate a result because no numbers were given';
+      }
+
+      let answer = numbers[0];
```
Comment (cschiragb) [inline calculator.js:16]: We could say `firstNumber` instead of `answer` here

File: calculator.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+          case '+':
+            answer = answer + number;
+            break;
+          case '-':
+            answer = answer - number;
+            break;
+          case '/':
+            answer = answer / number;
+            break;
+          default:
```
Comment (cschiragb) [inline calculator.js:35]: A case for `'*'` will need to be added

File: calculator.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+            console.log("Invalid operator " + operator);
+        }
+        // console.log('index: ' + index);
+        // console.log('current answer: ' + answer);
```
Comment (cschiragb) [inline calculator.js:39]: We can remove these unrequired comments :-)

File: index.js [hunk 4]
```diff
@@ -1,7 +1,20 @@
...
 console.log('Welcome to the calculator!');
 
-console.log('Please enter some input:');
-const response = readline.prompt();
-console.log('You entered: ' + response);
+console.log('Enter the operator:');
+const operator = readline.prompt();
+
+console.log(`How many numbers do you want to ${operator} ?`);
+const c = parseInt(readline.prompt());
```
Comment (cschiragb) [inline index.js:10]: `c` could be something more descriptive, such as `countOfNumbersToProcess`

File: index.js [hunk 5]
```diff
@@ -1,7 +1,20 @@
...
+
+const calculator = calc.create_calculator(operator);
+
+for (var i = 0; i < c; i++) {
+  console.log('Enter a number:');
+  const n = parseInt(readline.prompt());
```
Comment (cschiragb) [inline index.js:16]: `n` could be something more descriptive, such as `inputNumber`

File: index.js [hunk 6]
```diff
@@ -1,7 +1,20 @@
...
+  calculator.add_number(n);
+}
+
+console.log('The answer is: ' + calculator.calculate());
```
Comment (cschiragb) [inline index.js:20]: We can also write this as `console.log(``The answer is: ${calculator.calculate()}``};`

Comment (cschiragb) [reply index.js:20]: The text in the parentheses should be single backticks rather than double (the formatting of review comments doesn't allow this!)

File: calculator.js [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    calculate: function () {
+      // Return zero if there are no numbers to calculate
+      if (numbers.length === 0) {
+        throw 'Cannot calculate a result because no numbers were given';
+      }
+
+      let answer = numbers[0];
+
+      let i = 0;
+      numbers.forEach(function (number) {
```
Comment (cschiragb) [inline calculator.js:19]: We can use the number, index parameters for a `forEach `instead here, which saves us from having to define an i variable that we use to avoid operating on the first number unnecessarily. 

For example:

```
numbers.forEach(function (number, index) {
   if (index === 0) {
      return;
   }
   switch (operator) {
      ...
   }
}
```

File: test/calculator_test.js [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+  describe('#calculate()', function () {
+    it('adds one number', function () {
+      const calculator = calc.create_calculator('+');
+
+      calculator.add_number(16);
+
+      assert.equal(16, calculator.calculate());
+    });
+
+    it('adds two numbers', function () {
```
Comment (cschiragb) [inline test/calculator_test.js:15]: We don't need to test two numbers, as testing multiple numbers implicitly tests this anyway
We should however test one number as this is the edge case

File: test/calculator_test.js [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+      const calculator = calc.create_calculator('+');
+
+      calculator.add_number(5);
+      calculator.add_number(4);
+
+      assert.equal(9, calculator.calculate());
+    });
+
+    it('adds many numbers', function () {
```
Comment (cschiragb) [inline test/calculator_test.js:24]: We should test the other operations too (it would have picked up the missing multiplication operation in this case!)

File: test/calculator_test.js [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+      calculator.add_number(0);
+      calculator.add_number(-20);
+      calculator.add_number(1486);
+      calculator.add_number(-402);
+      calculator.add_number(4);
+
+      assert.equal(1077, calculator.calculate());
+    });
+
+    it('rejects the calculation when there are no numbers', function () {
```
Comment (cschiragb) [inline test/calculator_test.js:38]: We could also check for other invalid operations such as:
- A non-valid operator was provided
- A operator was provided
- A character instead of a number was provided
and any other edge cases...
